ipaserver/install/bindinstance.py

```diff
@@ -295,6 +295,28 @@ def find_reverse_zone(ip_address, api=api):
     return None
 
 
+def named_add_ext_conf_file(src, dest, gid, uid=0, mode=0o640, t_params={}):
+    """
+    Ensure ipa-ext.conf file is present, but don't override it.
+
+    :param src: String. Absolute path to source template
+    :param dest: String. Absolute path to destination
+    :param gid: Integer. Group id for destination file
+    :param uid: Integer. User id for destnation file. Defaults to 0 (root)
+    :param mode: String. Mode for destination file. Defaults to 0o640 (rw-r--)
+    :param t_params: Dict. Parameters for source template
+    """
+    if not os.path.exists(dest):
+        ipa_ext_txt = ipautil.template_file(src, t_params)
+        with open(dest, 'w') as ipa_ext:
+            ipa_ext.write(ipa_ext_txt)
+
+        os.chown(dest, uid, gid)
+        os.chmod(dest, mode)
```

Comment:
Please, use methods on file descriptors as I posted before.

Reply:
out of curiosity, does it change something regarding performances? Change already applied locally, I just want to know more :)

Reply:
I could name the authoritative source (at least, for me) `John Viega, Matt Messier Secure Programming Cookbook for C and C++` ISBN 0-596-00394-3.
Chapter `2.7 Restricting Access Permissions for New Files on Unix`.

Roughly speaking:
```python
mask = 0o027                                                                    
prev_umask = os.umask(mask)                                                     
with open(filepath, 'w') as f: 
    # for existing files only                                                
    os.fchmod(f.fileno(), 0o640)
    #                                               
    os.fchown(f.fileno(), 0, 0) 
    # write last                                              
    f.write(text)                                                               
os.umask(prev_umask) 
```